utils/Query.ts

```diff
@@ -103,69 +121,102 @@ async function getServer(id: string, topLevel=true): Promise<Server> {
 		.where({ id })
 		.orWhereRaw(`(flags & ${ServerFlags.trusted}) and vanity=?`, [id])
 		.orWhereRaw(`(flags & ${ServerFlags.partnered}) and vanity=?`, [id])
-	if (res[0]) {
-		const data = await getServerData(res[0].id)
-		if(!data || (+new Date() - +new Date(data.updatedAt)) > 3 * 60 * 1000) res[0].state = 'unreachable'
-		else {
-			res[0].flags = res[0].flags | (data.features.includes(GuildFeature.Partnered) && ServerFlags.discord_partnered) | (data.features.includes(GuildFeature.Verified) && ServerFlags.verified)
-			if(res[0].owners !== JSON.stringify([data.owner, ...data.admins]) || res[0].name !== data.name)
-				await knex('servers').update({ name: data.name, owners: JSON.stringify([data.owner, ...data.admins]) })
-					.where({ id: res[0].id })
+
+	if (!serverRow[0]) return null
+
+	const data = await getServerData(serverRow[0].id)
+
+	// Time elapsed since last update in milliseconds
+	const updateDiff = Date.now() - new Date(data.updatedAt).getTime()
+	if (!data || updateDiff > 3 * 60 * 1000) {
+		serverRow[0].state = 'unreachable'
+	} else {
+		const flags = {
+			partnered: data.features.includes(GuildFeature.Partnered) && ServerFlags.discord_partnered,
+			verified: data.features.includes(GuildFeature.Verified) && ServerFlags.verified,
 		}
-		delete res[0].owners
-		// console.log(data)
-		res[0].icon = data?.icon || null
-		res[0].members = data?.memberCount || null
-		res[0].emojis = data?.emojis || []
-		res[0].category = JSON.parse(res[0].category)
-		res[0].boostTier = data?.boostTier ?? null
-		if(topLevel) {
-			res[0].owner = await get._rawUser.load(data?.owner || '') ||  null
-			res[0].bots = (await Promise.all(data?.bots.slice(0, 3).map(el => get._rawBot.load(el)) || [])).filter(el => el) || null
-		} else {
-			res[0].owner = data?.owner || null
-			res[0].bots = data?.bots || null
+		serverRow[0].flags |= flags.partnered | flags.verified
+
+		const latestOwners = JSON.stringify([data.owner, ...data.admins])
+		if (serverRow[0].owners !== latestOwners || serverRow[0].name !== data.name) {
+			await knex('servers').update({ name: data.name, owners: latestOwners }).where({ id: serverRow[0].id })
 		}
 	}
-	return res[0] ?? null
+
+	// Delete deprecated fields
+	delete serverRow[0].owners
+
+	// FIXME: Type mismatch
+	let owner: any = data?.owner ?? null
+	let bots: any = data?.bots ?? null
+	if(topLevel) {
```

Comment:
this is the problem with typing, maybe you are okay with editing the type as User | string | null.